node/src/accountant/dao_utils.rs

```diff
@@ -343,15 +342,81 @@ pub fn sum_i128_values_from_table(
         .sum()
 }
 
+pub struct ThresholdUtils {}
+
+impl ThresholdUtils {
+    pub fn slope(payment_thresholds: &PaymentThresholds) -> i128 {
+        /*
+        Slope is an integer, rather than a float, to improve performance. Since there are
+        computations that divide by the slope, it cannot be allowed to be zero; but since it's
+        an integer, it can't get any closer to zero than -1.
+
+        If the numerator of this computation is less than the denominator, the slope will be
+        calculated as 0; therefore, .permanent_debt_allowed_gwei must be less than
+        .debt_threshold_gwei, so that the numerator will be no greater than -10^9 (-gwei_to_wei(1)),
+        and the denominator must be less than or equal to 10^9.
+
+        These restrictions do not seem over-strict, since having .permanent_debt_allowed greater
+        than or equal to .debt_threshold_gwei would result in chaos, and setting
+        .threshold_interval_sec over 10^9 would mean continuing to declare debts delinquent after
+        more than 31 years.
+
+        If payment_thresholds are ever configurable by the user, these validations should be done
+        on the values before they are accepted.
+        */
+
+        (gwei_to_wei::<i128, _>(payment_thresholds.permanent_debt_allowed_gwei)
+            - gwei_to_wei::<i128, _>(payment_thresholds.debt_threshold_gwei))
+            / payment_thresholds.threshold_interval_sec as i128
```

Comment:
Oh... the casting is going on after a different argument actually... it's just   `payment_thresholds.threshold_interval_sec as i128`. Which is safe because `threshold_interval_sec` is of type u64.

Reply:
not true...it works clearly...you can easily check what type `debt_threshold_gwei` is... for me the number would be just noise :man_shrugging: